python_modules/dagster-graphql/dagster_graphql/implementation/fetch_assets.py

```diff
@@ -196,20 +198,35 @@ def get_asset_nodes_by_asset_key(
 
     dynamic_partitions_loader = CachingDynamicPartitionsLoader(graphene_info.context.instance)
 
-    asset_nodes_by_asset_key: Dict[AssetKey, GrapheneAssetNode] = {}
+    asset_nodes_by_asset_key: Dict[
+        AssetKey, Tuple[CodeLocation, ExternalRepository, ExternalAssetNode]
+    ] = {}
     for repo_loc, repo, external_asset_node in asset_node_iter(graphene_info):
-        preexisting_node = asset_nodes_by_asset_key.get(external_asset_node.asset_key)
-        if preexisting_node is None or preexisting_node.external_asset_node.is_source:
-            asset_nodes_by_asset_key[external_asset_node.asset_key] = GrapheneAssetNode(
+        preexisting_node_tuple = asset_nodes_by_asset_key.get(external_asset_node.asset_key)
+        if preexisting_node_tuple is None or preexisting_node_tuple[2].is_source:
```

Comment:
nit: indexing into tuples creates confusing code.

```suggestion
        preexisting_external_asset_node = asset_nodes_by_asset_key.get(external_asset_node.asset_key, (None, None, None)
        if preexisting_external_asset_node and  preexisting_external_asset_node.is_source:
```